### PR TITLE
Checking if MPI is enabled to allow both Trilinos and PETSc.

### PR DESCRIPTION
**Pull request by fmilicchio**
Dear all,

I am not skilled in large CMake projects, but I've done my best to check the incompatibility between PETSc + Trilinos and not allowing MPI.

I hope this helps.

Cheers,
    Franco

### CODE REVIEW

**Comment by drwells** [comment]
This looks good!

I have a minor request: could you extend the error message to mention that the incompatibility stems from Trilinos and PETSc providing mutually incompatible MPI stubs?

**Comment by drwells** [comment]
Fixes #8366.

**Comment by fmilicchio** [comment]
@drwells I hope you can see this last commit, as requested.

Thanks!

File: cmake/configure/configure_2_trilinos.cmake [hunk 1]
```diff
@@ -103,6 +103,27 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
       SET(${var} FALSE)
     ENDIF()
 
+    #
+    # deal.II has to be configured with MPI if both Trilinos and PETSc are
+    # enabled.
+    #
+    IF( DEAL_II_WITH_TRILINOS AND DEAL_II_WITH_PETSC AND NOT DEAL_II_WITH_MPI )
```
Comment (tamiko) [inline cmake/configure/configure_2_trilinos.cmake]: Would you mind removing the spaces between the argument and the opening and closing parenthesis?

File: cmake/configure/configure_2_trilinos.cmake [hunk 2]
```diff
@@ -103,6 +103,27 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
...
+	"provide mutually incompatible MPI stubs."
+        )
+      SET(TRILINOS_ADDITIONAL_ERROR_STRING
+        ${TRILINOS_ADDITIONAL_ERROR_STRING}
+        "To enable both Trilinos and PETSc, deal.II must be configured with MPI "
+        "or enable one between Trilinos and PETSc removing MPI. Both Trilinos and PETSc "
+	"provide mutually incompatible MPI stubs, but found:\n"
+        "  DEAL_II_WITH_MPI = ${DEAL_II_WITH_MPI}\n"
+        "  DEAL_II_WITH_TRILINOS = ${DEAL_II_WITH_TRILINOS}\n"
+        "  DEAL_II_WITH_PETSC = ${DEAL_II_WITH_PETSC}\n"
```
Comment (tamiko) [inline cmake/configure/configure_2_trilinos.cmake]: What about
```
     SET(TRILINOS_ADDITIONAL_ERROR_STRING
        ${TRILINOS_ADDITIONAL_ERROR_STRING}
        "Incompatible Trilinos and PETSc libraries found. Both libraries were"
        "configured without MPI support and cannot be used at the same time due"
        "to incompatible mpi stub files. Either reconfigure deal.II, Trilinos,"
        and PETSc with MPI support, or disable one of the libraries.\n"
        )
```

Comment (tamiko) [reply cmake/configure/configure_2_trilinos.cmake]: I think this makes the problem (and what one has to do) a bit clearer.

Comment (fmilicchio) [reply cmake/configure/configure_2_trilinos.cmake]: @tamiko I've included your changes and removed the spurious spaces in the IF() call.

**Comment by bangerth** [comment]
@tamiko ?

**Comment by masterleinad** [comment]
@tamiko ping

**Comment by tamiko** [comment]
@bangerth Ups. You can also "dismiss" the stale review in order to merge :-)

**Pull request merged**